doc/rst/tools/mason/mason.rst

```diff
@@ -159,202 +194,364 @@ To try out different values at runtime, pass the values for ``number`` to ``maso
 
-For example, ``Mason.toml``:
+  Test Passed!
 
-.. code-block:: text
+  --------------------
 
-    [brick]
-    name = "MyPackage"
-    version = "0.1.0"
-    chplVersion = "1.16.0"
-    authors = ["Sam Partee <Sam@Partee.com>"]
+  --- Results ---
+  Test: myProjectTest Passed
 
-    [dependencies]
-    curl = '1.0.0'
+  --- Summary:  1 tests run ---
+  -----> 1 Passed
+  -----> 0 Failed
 
-The ``chplVersion`` field indicates Chapel releases compatible with this
-package. There are a number of accepted formats:
+Mason will find tests either by searching through the ``test/`` directory, or by
+reading them from the ``Mason.toml`` where they can be specified.
 
-.. code-block:: text
 
-    "1.16.0"         # 1.16.0 or later
-    "1.16"           # 1.16.0 or later
-    "1.16.0..1.19.0" # 1.16 through 1.19, inclusive
+Creating and Running Examples
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-By default, ``chplVersion`` is set to represent the current Chapel release or
-later. For example, if you are using the 1.16 release, chplVersion will be
-``1.16.0``.
+Examples are a great way of showing off what a project can do. Examples are easy
+to run and configure. The following example adds an example to ``MyProject`` and
+runs it. The example added simply prints a number of times based on a config const as follows:
 
 
-TOML
-====
+.. code-block:: chpl
 
-TOML is the configuration language chosen by the chapel team for
-configuring programs written in chapel. A TOML file contains the
-necessary information to build a chapel program using mason.
-`TOML Spec <https://github.com/toml-lang/toml>`_.
+   config const count: int = 10;
 
+   for i in 0..count {
+     writeln("This is an example!!");
+   }
 
-Mason-Registry
-==============
 
-The default mason registry is a GitHub repository containing a list of versioned manifest files.
+To build the example without running it, use the command ``mason build --example``.
+This command will build ALL examples found either in the ``example/`` directory or
+listed in the Mason.toml.
 
-`Mason-Registry <https://github.com/chapel-lang/mason-registry>`_.
+.. note:: If examples or tests are listed in the Mason.toml, Mason will not search
+          for any examples or tests not listed.
 
-A registry will be downloaded to ``$MASON_HOME/<name>`` by ``mason update``
-for each registry named in ``$MASON_REGISTRY`` if a registry at that location
-does not already exist.
+To view what examples are available, enter ``mason run --example`` without any other
+arguments. This will produce the names of all examples that are currently available
+to Mason::
 
-The registry consists of a hierarchy like the following:
+  --- available examples ---
+  --- myProjectExample.chpl
+  --------------------------
+
+To run the example, use the command ``mason run --example myProjectExample.chpl``.
+
+After the program is run via the command above, the project structure will look as
+follows::
+
+
+  MyProject
```

Comment:
May be easier to distinguish directories from files by adding / at the end of directories.